### PR TITLE
fix(DEV-1120): Move PR Title workflow to workflow template

### PR DESCRIPTION
**Pull request by jazeee**
Summary
===

Github workflow templates can be used to define organization-wide
workflows.
See:
<https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization>

We want to ensure that PR Titles meet our guidelines.

See: DEV-1113

Add GitHub action to validate PR title to meet semantic versioning patterns (#3)

The semantic versioning relies on the title of the commit to determine
whether the next semantic version release.

This change is intended to solve the problem that `commitlint` attempts
to solve.
Unfortunately, `commitlint` attempts to sanitize commit messages, before
the PR is created or updated.
It doesn't work, and encourages bad patterns and workarounds like
using a nonsensical commit message like:
`fix: Ab cd` or `docs: He ll`

Worse, it doesn't allow us to add useful context like long URLs.
Reference:

<https://www.notion.so/0019-Commit-Messages-b7cc6f56354f4f28b902df7cd13b2450>

In addition, people don't typically review the commit message, nor
does anyone fix the commit message content. The commit messages remain
uncurated.

In contrast, the PR title and description are read by reviewers, and are
more
likely to be curated.
In addition, since the PR title and description is better curated,
the PR makes sense as the source of truth for the bolus of change.
In other words, it is the best piece of information to drive the commit
log.

Reference:

<https://linear.app/clipboardhealth/issue/DEV-1113/add-semantic-version-github-action-to-cbh-core>

Ticket:
<https://linear.app/clipboardhealth/issue/DEV-1120/move-pr-title-workflow-to-workflow-template>

Changes
===

- Move github action to check the PR title to match github documentation
- Add README

### CODE REVIEW

File: .github/workflow-template/pull-request-title.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
       - unlabeled
```
Comment (pgbezerra) [inline .github/workflow-template/pull-request-title.yml]: I think the template folder should be in the root

Comment (jazeee) [reply .github/workflow-template/pull-request-title.yml]: You are right. I missed that.
Fixed.

**Comment by jazeee** [comment]
From what I can tell, the template doesn't seem to affect this repo. 🤷🏾 
Unsure how it will work across repos.

**Review by pgbezerra — APPROVED** [review]
